samples/webforms/embed-a-dashboard-into-an-app/PBIWebApp/Default.aspx.cs

```diff
@@ -0,0 +1,189 @@
+                {"client_id", Properties.Settings.Default.ClientID},
+
+                //Resource uri to the Power BI resource to be authorized
+                {"resource", "https://analysis.windows.net/powerbi/api"},
+
+                //After user authenticates, Azure AD will redirect back to the web app
+                {"redirect_uri", "http://localhost:13526/Redirect"}
+            };
+            
+            //Create sign-in query string
+            var queryString = HttpUtility.ParseQueryString(string.Empty);
+            queryString.Add(@params);
+
+            //Redirect authority
+            //Authority Uri is an Azure resource that takes a client id to get an Access token
+            string authorityUri = "https://login.windows.net/common/oauth2/authorize/";
+            var authUri = String.Format("{0}?{1}", authorityUri, queryString);
+            Response.Redirect(authUri);       
+        }
+
+        protected void getDashboardsButton_Click(object sender, EventArgs e)
+        {
+            string responseContent = string.Empty;
+
+            //Configure dashboards request
+            System.Net.WebRequest request = System.Net.WebRequest.Create(String.Format("{0}dashboards", baseUri)) as System.Net.HttpWebRequest;
+            request.Method = "GET";
+            request.ContentLength = 0;
+            request.Headers.Add("Authorization", String.Format("Bearer {0}", authResult.AccessToken));
+
+            //Get dashboards response from request.GetResponse()
+            using (var response = request.GetResponse() as System.Net.HttpWebResponse)
+            {
+                //Get reader from response stream
+                using (var reader = new System.IO.StreamReader(response.GetResponseStream()))
+                {
+                    responseContent = reader.ReadToEnd();
+
+                    //Deserialize JSON string
+                    PBIDashboards PBIDashboards = JsonConvert.DeserializeObject<PBIDashboards>(responseContent);
+                    Table1.Visible = true;
+                    //Get each dashboard 
+                    foreach (PBIDashboard dsb in PBIDashboards.value)
+                    {
+                        TableRow tRow = new TableRow();
+
+                        Table1.Rows.Add(tRow);
+                        TableCell idCell = new TableCell();
+                        idCell.Text = dsb.id;
+                        tRow.Cells.Add(idCell);
+
+                        Table1.Rows.Add(tRow);
+                        TableCell nameCell = new TableCell();
+                        nameCell.Text = dsb.displayName;
+                        tRow.Cells.Add(nameCell);
+
+                        Table1.Rows.Add(tRow);
+                        TableCell isReadOnlyCell = new TableCell();
+                        isReadOnlyCell.Text = dsb.isReadOnly.ToString();
+                        tRow.Cells.Add(isReadOnlyCell);
+
+                        Table1.Rows.Add(tRow);
```

Comment:
I think that you add the same row 4 times for each dashboard.
you have: Table1.Rows.Add(tRow); 4 times in each loop.